tests/rptest/tests/end_to_end.py

```diff
@@ -203,6 +203,11 @@ def on_record_consumed(self, record, node):
         self.records_consumed.append((key, record_id))
 
     def await_consumed_offsets(self, last_acked_offsets, timeout_sec):
+        # if consumer was interrupted with error there is no point waiting
+        # for it to finish
+        if self.consumer.interrupted_with_error:
```

Comment:
q: this can go inside has_finished_consuming()? Incase it is set after we enter has_finished_consuming loop.